src/mb_enhanced_cover_art_uploads/form.ts

```diff
@@ -24,16 +25,21 @@ async function enqueueImage(image: FetchedImage, defaultTypes: ArtworkTypeIDs[],
 function dropImage(imageData: File): void {
     // Fake event to trigger the drop event on the drag'n'drop element
     // Using jQuery because native JS cannot manually trigger such events
-    // for security reasons
-    const dropEvent = $.Event('drop') as JQuery.TriggeredEvent;
-    dropEvent.originalEvent = {
+    // for security reasons.
+    const dropEvent = getFromPageContext('$').Event('drop') as JQuery.TriggeredEvent;
+    // We need to clone the underlying data since we might be running as a
+    // content script, meaning that even though we trigger the event through
+    // unsafeWindow, the page context may not be able to access the event's
+    // properties.
+    dropEvent.originalEvent = cloneIntoPageContext({
         dataTransfer: { files: [imageData] }
-    } as unknown as DragEvent;
+    } as unknown as DragEvent);
 
     // Note that we're using MB's own jQuery here, not a script-local one.
-    // We need to reuse MB's own jQuery to be able to trigger the event
-    // handler.
-    $('#drop-zone').trigger(dropEvent);
+    // We cannot manually dispatch the event due to security reasons, so we
+    // use MB's own jQuery to find the event handler and call it manually.
+    // @ts-expect-error: _data is not defined in declaration
+    getFromPageContext('$')._data(qs('#drop-zone'), 'events').drop[0].handler(dropEvent);
```

Comment:
Many of the other changes look acceptable for the advantage of a wider compatibility but this one looks just ugly 🤮

Reply:
I'm actually not 100% sure it's truly necessary. Some change _is_ definitely necessary, but the main problem with `$("#drop-zone").trigger(dropEvent)` is that jQuery tries to call stuff like `isDefaultPrevented()` on the event, which is a function in the content context and leads to a permission error. However, `cloneInto()` can also clone functions into the page context, which I believe I forgot to set as a parameter, so perhaps enabling that will make it work… Will check